x-pack/plugins/monitoring_collection/server/lib/get_kibana_stats.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import { Observable } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { ServiceStatus, ServiceStatusLevels } from '../../../../../src/core/server';
+
+const SNAPSHOT_REGEX = /-snapshot/i;
+
+const ServiceStatusToLegacyState: Record<string, string> = {
+  [ServiceStatusLevels.critical.toString()]: 'red',
+  [ServiceStatusLevels.unavailable.toString()]: 'red',
+  [ServiceStatusLevels.degraded.toString()]: 'yellow',
+  [ServiceStatusLevels.available.toString()]: 'green',
+};
+
+export async function getKibanaStats({
+  config,
+  overallStatus$,
+}: {
+  config: {
+    allowAnonymous: boolean;
+    kibanaIndex: string;
+    kibanaVersion: string;
+    uuid: string;
+    server: {
+      name: string;
+      hostname: string;
+      port: number;
+    };
+  };
+  overallStatus$: Observable<ServiceStatus>;
+}) {
+  const overallStatus = await overallStatus$.pipe(first()).toPromise();
```

Comment:
I recommend subscribing to `overallStatus$` within `server/plugin` and having it update a variable at some interval.


```
    let status: ServiceStatus<unknown>;
    core.status.overall$.subscribe((newStatus) => {
      status = newStatus;
    });
    const getStatus = () => {
      return status;
    };
```

Then, this function could use the `getStatus` function, and be converted into a synchronous function.

Reply:
Sure, sounds good